src/components/ViewportButton/index.tsx

```diff
@@ -53,6 +55,11 @@ const ViewportButton: React.FC<ViewportButtonProps> = ({
         }
     };
 
+    const getTooltip = () =>
+        !disabled || tooltipWhenDisabled ? tooltipText : "";
+
+    const getClickHandler = () => (!disabled ? clickHandler : undefined);
```

Comment:
I'm surprised typescript isn't upset about having `onClick={undefined}`. 
Just a suggestion: 
```suggestion
    onClick = () => {
       if (disabled) {return} else {clickHander()}
    }
    
    onClick={onClick}
    
```
the only difference being that it's now a function that returns nothing rather than setting it to undefined. but it's fine either way.

Reply:
Returning `undefined` seems fine to me, but I might consider making these just plain constants.